src/core/Tree.tsx

```diff
@@ -236,7 +236,10 @@ function renderItem(el: IAnyStateTreeNode, includeKey = true) {
  */
 function renderChildren(item: IAnyStateTreeNode) {
   if (item && item.children && item.children.length) {
+    const defaultToName = item.children[0].isObject ? item.children[0].name : item.toname;
```

Comment:
we should look for the first object in all the config, not in one level of controls we can fix. also this `toname` will be `undefined` in most of the cases (when the first tag is `Header` for example), so all the controls inside this `item` will be ruined.